addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1006,6 +1007,34 @@ return AbstractRenderer.extend({
         }
         return shortLabel;
     },
+    /**
+     * sort datapoints as per given order(Ascending or Descending)
+     *
+     * @private
+     * @param {Object[]} dataPoints
+     * @returns {Object[]} sorted dataPoints if orderby set on state
+     */
+    _sortDataPoints(dataPoints) {
+        if (!_.size(this.state.timeRanges) && this.state.orderBy && ['bar', 'line'].includes(this.state.mode)) {
+            /* group data with label[0](i.e. groupby[0] which displayed on x-axis)
+            and then sort datapoints based on sum of value in ascending/descending
+            order */
+            const groupedDataPoints = {};
+            dataPoints.forEach(function (datapoint) {
+                if (!groupedDataPoints[datapoint.labels[0]]) {
+                    groupedDataPoints[datapoint.labels[0]] = [];
+                }
+                groupedDataPoints[datapoint.labels[0]].push(datapoint);
+            });
+            dataPoints = _.sortBy(groupedDataPoints, function (datapoint) {
```

Comment:
there is a sortBy function in web.utils

Reply:
Hello @aab-odoo 

I have checked with sortBy method of utils, But it is not working here
it will work only when we don't have grouped data.
But for sorting of grouped data we are creating the dict of grouped data for each group
and then sort it with the sum of it's values

_.sortBy method supports the use of dict but utils.sortBy is not supporting dict
it expects the list from us

So, I have done this changes without using underscore 
 https://github.com/odoo/odoo/pull/49970/commits/12a6690e0ed317c6390a2818a94f0722c0249526#diff-1eb60ace245345929e24d244c1c86197R1023
Can you please check it?

Thanks